### PR TITLE
add output result

### PR DESCRIPTION
**Pull request by soulseen**
Signed-off-by: soulseen <sunzhu@yunify.com>

### CODE REVIEW

File: pkg/api/types.go [hunk 1]
```diff
@@ -267,6 +267,11 @@ type Config struct {
 
 	// The RevisionId is a branch name or a SHA-1 hash of every important thing about the commit
 	RevisionId string `json:"revisionId,omitempty"`
+
+	// Output build result. If build not in k8s cluster, can not use this field.
+	OutputBuildResult bool `json:"outputBuildResult,omitempty"`
+	// S2irun job's name and namespace.
+	NamespacedName NamespacedName `json:"namespacedName,omitempty"`
```
Comment (runzexia) [inline pkg/api/types.go]: * NamespacedName

File: pkg/outputresult/output.go [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+	// add Annotation
+	err := addBuildResultToAnnotation(result)
+	if err != nil {
+		glog.V(1).Info("Output build result failed, reason: %s.", err)
+	}
+}
+
+func addBuildResultToAnnotation(buildResult *api.Result) error {
+	namespace := os.Getenv(S2irunNamespace)
```
Comment (runzexia) [inline pkg/outputresult/output.go]: why use env here?  use api.config?

File: pkg/outputresult/output.go [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+	"k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/util/retry"
+	"os"
+	"sigs.k8s.io/controller-runtime/pkg/client/config"
+)
+
+const (
+	AnnotationBuildResultKey = "s2iBuildResult"
+	AnnotationBuildSourceKey = "s2iBuildSource"
```
Comment (runzexia) [inline pkg/outputresult/output.go]: These const could be moved to apis

File: pkg/api/types.go (hunk 1 above)
Comment (soulseen) [reply pkg/api/types.go]: NamespacedName not used, will remove.

File: pkg/outputresult/output.go (hunk 2 above)
Comment (soulseen) [reply pkg/outputresult/output.go]: NamespacedName not used, will remove.

File: pkg/outputresult/output.go (hunk 3 above)
Comment (soulseen) [reply pkg/outputresult/output.go]: done

File: pkg/api/types.go [hunk 4]
```diff
@@ -43,8 +43,24 @@ const (
...
+	AnnotationBuildSourceKey = "s2iBuildSource"
+
+	// Command for pull docker image.
+	CommandPull = "docker pull "
+
+	// Default source branch.
+	DefaultBranch = "master"
+
+	//S2irun Namespace which s2i job run.
+	S2irunNamespace = "S2irunNamespace"
```
Comment (runzexia) [inline pkg/api/types.go]: Kind is S2iRun

File: pkg/outputresult/output.go [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+	if err != nil {
+		glog.V(1).Info("Output build result failed, reason: %s.", err)
+	}
+}
+
+func addBuildResultToAnnotation(buildResult *api.Result) error {
+	namespace := os.Getenv(api.S2irunNamespace)
+	jobName := os.Getenv(api.S2irunJobName)
+	if namespace == "" || jobName == "" {
+		glog.Errorf("failed to get env S2irunJobName and S2irunNamespace")
```
Comment (runzexia) [inline pkg/outputresult/output.go]: warn or info ?

Comment (soulseen) [reply pkg/outputresult/output.go]: warn is better.

**Pull request merged**